modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/ServiceConfigurationManager.java

```diff
@@ -85,42 +112,118 @@ public ServiceOrder changeServiceConfiguration(String serviceId,
+        List<ServiceConfigurationParameter> configurationParameters =
+                ocl.getServiceConfigurationManage().getConfigurationParameters();
+
+        List<ServiceConfigurationUpdateRequest> requests = new ArrayList<>();
+        deployResourceMap.forEach((groupName, deployResourceList) -> {
+            configManageScripts.forEach(configManageScript -> {
+                if (configManageScript.getConfigManager().equals(groupName)) {
+                    if (!CollectionUtils.isEmpty(deployResourceList)) {
+                        Map<String, Object> properties =
+                                getServiceConfigurationUpdateProperties(groupName,
+                                        configurationParameters, updateRequestMap);
+                        if (configManageScript.getRunOnlyOnce()) {
+                            ServiceConfigurationUpdateRequest request =
+                                    getInitServiceConfigurationUpdateRequest(orderId, groupName,
+                                            deployServiceEntity, properties);
+                            request.setResourceName(
+                                    deployResourceList.getFirst().getResourceName());
+                            requests.add(request);
+                        } else {
+                            deployResourceList.forEach(deployResource -> {
+                                ServiceConfigurationUpdateRequest request =
+                                        getInitServiceConfigurationUpdateRequest(orderId, groupName,
+                                                deployServiceEntity, properties);
+                                request.setResourceName(deployResource.getResourceName());
+                                requests.add(request);
+                            });
+
+                        }
+                    }
+                }
+            });
+        });
+
+        if (!CollectionUtils.isEmpty(requests)) {
+            serviceConfigurationUpdateStorage.batchInsert(requests);
+        }
+    }
+
+    private Map<String, Object> getServiceConfigurationUpdateProperties(String groupName,
+            List<ServiceConfigurationParameter> params, Map<String, Object> updateRequestMap) {
+
+        Map<String, Object> existsServiceConfig = new HashMap<>();
+        params.forEach(serviceConfigurationParameter -> {
+            if (groupName.equals(serviceConfigurationParameter.getManagedBy())) {
+                existsServiceConfig.put(serviceConfigurationParameter.getName(),
+                        serviceConfigurationParameter.getInitialValue());
+            }
+        });
+        updateRequestMap.forEach((k, v) -> {
+            if (existsServiceConfig.containsKey(k)) {
+                existsServiceConfig.put(k, v);
+            }
+        });
+        return existsServiceConfig;
+    }
+
+    private ServiceConfigurationUpdateRequest getInitServiceConfigurationUpdateRequest(UUID orderId,
```

Comment:
we can remove the 'init' word from this method name as it is not the initial request only.